docs/change-log/index.md

```diff
@@ -13,16 +14,183 @@
+Starting with v1.3.0, and continuing in subsequent releases, obsolete functions are getting removed from the code.
+
+**Deprecated and removed from v1.3.0 code**<br>
+These MLRun APIs have been deprecated since at least v1.0.0 and were removed from the code:
+
+| Deprecated/removed                   | Use instead                                   |
+| ------------------------------------ | --------------------------------------------- |
+| `project.workflows`                  | `project.spec.workflows`, such as `get_or_create_project`, `load_project` |
+| `project.functions`                  | `project.spec.functions`                      |
+| `project.artifacts`                  | `project.spec.artifacts`                       |
+| `project.func()`                     | `project.get_function()`                       |
+| `project.create_vault_secrets()`     | NA                                   |
+| `project.get_vault_secret()`         | NA                                   |
+| `MlrunProjectLegacy` class           | `MlrunProject`                                 |
+| Feature-store: usage of state in graph. For example: `add_writer_state`, and the `after_state` parameter in `_init_` methods.  | `step`                        |
+| `mount_path` parameter in mount_v3io() | `volume_mounts`                        |
+| `NewTask`                            | `run_function()`                        |
+
+
+
+**Deprecated APIs, will be removed in v1.5.0**<br>
+These APIs will be removed from the v1.5.0 code. A FutureWarning appears if you try to use them in v1.3.0.
+| Deprecated / to be removed                       | Use instead                                   |
+| ------------------------------------------------ | --------------------------------------------- |
+| project-related parameters of `set_environment`. (Global-related parameters will not be deprecated.) | Project APIs 
+such as `get_or_create_project`, `load_project` |
+| `KubeResource.gpus`                              | `with_limits`                 |
+| Dask `gpus`                                      | `with_scheduler_limits` / `with_worker_limits`   |
+| Dask `with_limits`                               | `with_scheduler_limits` / `with_worker_limits`    |
+| Dask `with_requests`                             | `with_scheduler_requests` / `with_worker_requests`    |
+| Spark runtime `gpus`                              | `with_driver_limits` / `with_executor_limits` |
+| `mount_v3io_legacy` (mount_v3io no longer calls it) | `mount_v3io`                       |
+| `mount_v3io_extended`                            | `mount_v3io`                   |
+| `LegacyArtifact` and all legacy artifact types that inherit from it (full list is: `LegacyArtifact`, `LegacyDirArtifact`, `LegacyLinkArtifact`, `LegacyPlotArtifact`, `LegacyChartArtifact`, `LegacyTableArtifact`, `LegacyModelArtifact`, `LegacyDatasetArtifact`, `LegacyPlotlyArtifact`, `LegacyBokehArtifact`) | `Artifact`                       |
```

Comment:
If you chose to place the full list of deprecated classes, then it makes sense to note in the "use instead" that you should use "`Artifact` or other artifact classes that inherit from it". We don't want to generate an appearance that all classes mentioned are replaced by the single `Artifact` class.